RetailCoder.VBE/Root/RubberduckIoCInstaller.cs

```diff
@@ -198,11 +220,21 @@ private static void ApplyDefaultInterfaceConvention(IWindsorContainer container,
             }
         }
 
-        private static void RegisterFactories(IWindsorContainer container, Assembly[] assembliesToRegister)
+        private void RegisterFactories(IWindsorContainer container, Assembly[] assembliesToRegister)
         {
             foreach (var assembly in assembliesToRegister)
             {
-                container.Register(Types.FromAssembly(assembly)
+                var assemblyTypes = assembly.DefinedTypes
+                    .Where(w =>
+                    {
+                        var attribute = w.CustomAttributes.FirstOrDefault(f => f.AttributeType == typeof(ExperimentalAttribute));
+                        var ctorArg = attribute?.ConstructorArguments.Any() == true ? (string)attribute.ConstructorArguments.First().Value : string.Empty;
+
+                        return attribute == null ||
+                               _initialSettings.EnableExperimentalFeatures.Any(a => a.Key == ctorArg && a.IsEnabled);
+                    });
```

Comment:
Would it be useful to extract the construction of "available" assemblyTypes into a method?

Reply:
Probably.